src/app/core/auth-guard/auth.guard.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  RouterStateSnapshot,
+  CanActivate,
+  CanLoad,
+  Router,
+  Route,
+  UrlSegment,
+} from '@angular/router';
+import { Observable } from 'rxjs';
+import { tap, take, map } from 'rxjs/operators';
+
+import { AuthService } from '../../auth/auth.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard implements CanActivate, CanLoad {
+  constructor(
+    private router: Router,
+    private authService: AuthService,
+  ) {}
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot) {
+    return this.isAuthenticated().pipe(
+      tap(isAuthenticated => isAuthenticated || this.router.parseUrl('/auth/login')),
+    );
+  }
+
+  canLoad(route: Route, segments: UrlSegment[]) {
+    return this.isAuthenticated().pipe(
+      tap(isAuthenticated => isAuthenticated || this.router.navigate(['/auth', 'login'])),
+    );
+  }
+
+  isAuthenticated(): Observable<boolean> {
+    return this.authService.user$.pipe(
+      take(1),
```

Comment:
I think it's not necessary here because each navigation `canActivate` recreates subscription again and unsubscribe after.

Reply:
No, it doesn't work so